packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -202,9 +226,205 @@ export default async function initSanity(
   // Ensure we are using the output path provided by user
   outputPath = answers.outputPath
 
+  if (initNext) {
+    const useTypeScript = await prompt.single<boolean>({
+      type: 'confirm',
+      message: `Would you like to use TypeScript?`,
+      default: true,
+    })
+
+    const fileEnding = useTypeScript ? 'ts' : 'js'
+
+    const embeddedStudio = await prompt.single({
+      type: 'confirm',
+      message: `Would you like an embedded Sanity Studio?`,
+      default: true,
+    })
+
+    if (embeddedStudio) {
+      const useAppDir = await prompt.single({
+        type: 'confirm',
+        message: `Would you like to use the Next.js app dir?`,
+        default: false,
+      })
+
+      // find source path (app or pages dir)
+      const srcDir = useAppDir ? 'app' : 'pages'
+      let srcPath = path.join(workDir, srcDir)
+
+      if (!existsSync(srcPath)) {
+        srcPath = path.join(workDir, 'src', srcDir)
+        if (!existsSync(srcPath)) {
+          await fs
+            .mkdir(srcPath, {recursive: true})
+            .catch(() => debug('Error creating folder %s', srcPath))
+        }
+      }
+
+      const studioPath = await prompt.single({
+        type: 'input',
+        message: 'What url do you want to use for the embedded studio?',
+        default: '/studio',
+        validate(input) {
+          if (!input.startsWith('/')) {
+            return 'Must start with /'
+          }
+
+          if (input.endsWith('/')) {
+            return 'Must not end with /'
+          }
+
+          // a-Z, 0-9, -, _ and /
+          if (!/^[a-zA-Z0-9-_\\/]+$/.test(input)) {
+            return 'Must only contain a-Z, 0-9, -, _ and /'
+          }
+
+          return true
+        },
+      })
+
+      const embeddedStudioPath = path.join(
+        srcPath,
+        `${studioPath}/`,
+        useAppDir ? '[[..index]]/page.tsx' : '[[...index]].tsx'
+      )
+
+      await writeOrOverwrite(
+        embeddedStudioPath,
+        (useAppDir ? sanityStudioAppTemplate : sanityStudioPagesTemplate).replace(
+          ':configPath:',
+          new Array(embeddedStudioPath.slice(workDir.length).split('/').length - 1)
+            .join('../')
+            .concat('sanity.config')
+        )
+      )
```

Comment:
Can you add a small comment explaining what this is doing / with the goal?
I think I get the general gist of it (being related to the experimental App directory) but it would be wiser to keep a comment about it

Reply:
1. If finds out whether you're using `app` or `pages` dir and then chooses the correct template string
2. It replaces the `:configPath:` string from the template string with the correct path to `sanity.config.ts` which is in the root. Because `embeddedStudioPath` can be manually written by the user (default is `/studio`) but user can mount it anywhere, e.g. `/my/deeply/nested/studio` we have to account for the nesting when importing the config.
The `new Array(...)` creates enough `../` to reach the root level.

Reply:
Thanks for the explanation! I originally meant to add a comment on the code itself with a short explanation so that it doesn't become a question mark when other people stumble upon it (outside of this PR :) ), if you could do that it would be great

Reply:
Oh 🤦🏼‍♂️ of course! will do